dtl-lexer/src/variable.rs

```diff
@@ -354,7 +308,18 @@ mod tests {
             .iter()
             .map(|t| match t {
                 Ok(t) => match t.argument {
-                    Some(ref a) => (t.content(template), Some(a.content(template))),
+                    Some(ref a) => match a {
+                        Argument::Numeric(num) => {
+                            (t.content(template), Some(num.to_string().as_str()))
+                        }
+                        Argument::Text(text) => (t.content(template), Some(text.as_str())),
+                        Argument::TranslatedText(text) => {
+                            (t.content(template), Some(text.as_str()))
+                        }
+                        Argument::Variable(var) => {
+                            (t.content(template), Some(template.content(var.at)))
+                        }
+                    },
```

Comment:
So this creates a compilation error. As I understand, I return a `&str` that is deallocated at the end of the scope but I don't know how to solve this. There seems to be no way to force a lifetime onto a pointer. I'm curious how to solve this.

Reply:
You've rediscovered a reason why we use the current design of returning a `(usize, usize)` (`at`) in the lexer stage. This avoids needing to store references with lifetimes or needing extra allocations for owned values.

Reply:
Ok then I think I can find a solution if, instead of passing `Chars` to `lex_*`, it requires an `&'t str` slice?